src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift

```diff
@@ -231,66 +161,222 @@ extension Service {
 	///   - resource: Generic ("R") object and normally of type ReceiveResource.
 	///   - error: The error that would be thrown with the fail.
 	///   - completion: Swift-Result of loading. If successful, it contains the concrete object of our call.
-	func failureOrDefaultValueHandling<R>(
+	func retryOrDefaultValueOrFailureHandling<R>(
 		_ resource: R,
 		_ error: ServiceError<R.CustomError>,
-		_ responseData: Data? = nil
-	) -> Result<R.Receive.ReceiveModel, ServiceError<R.CustomError>> where R: Resource {
-		// Check if we have default value. If so, return it independent wich error we had
-		if let defaultModel = resource.defaultModel {
-			Log.info("Found some default value", log: .client)
-			guard var modelWithMetadata = defaultModel as? MetaDataProviding else {
-				return .success(defaultModel)
-			}
-			Log.info("Found a defaultModel which conforms to MetaDataProviding. Adding metadata now.", log: .client)
-			modelWithMetadata.metaData.loadedFromDefault = true
-			guard let originalModelWithMetadata = modelWithMetadata as? R.Receive.ReceiveModel else {
-				Log.warning("Cast back to R.Receive.ReceiveModel failed. Returning the model without metadata.", log: .client)
-				return .success(defaultModel)
+		_ responseData: Data? = nil,
+		_ completion: @escaping (Result<R.Receive.ReceiveModel, ServiceError<R.CustomError>>) -> Void
+	) where R: Resource {
+
+		if resource.retryingCount > 1 {
```

Comment:
Shouldn't it be > 0 ?

Reply:
No, otherwise we would count the 0 -loop as a retry too. Then we would change all retry counts from 3 to 2. But here I wanted to keep the old logic to not confuse everyone, where we had 3 retries and it was declared as a "3".